geth/mailservice/mailservice_api.go

```diff
@@ -27,12 +36,21 @@ var (
 
 // PublicAPI defines a MailServer public API.
 type PublicAPI struct {
```

Comment:
Not yours, but doesn't it makes sense to call it just `API`?

Reply:
noticed that it is common to separate PrivateAPI and PublicAPI in geth, even if it is unlikely that we will have Private in future I would leave it as is

Reply:
yeah, but private API shouldn't leave the package theoretically.
To my taste I prefer more something like that:
```type api struct { ... }``` for private and ```type API interface { ... }``` for the public APIs.

then it would be referenced as `mailserver.API` in the other places.

that doesn't matter that much though :)

Reply:
My 2 cents on this: `PublicAPI` gives an immediate hint that this will be used for geth's RPC subsystem. Naming `Public<Something>API` is common for geth code, so this name is actually helpful for the reader to understand what kind of "API" is this.

Reply:
Should non-public API ever be exported from the package?
My take on it was that if it is exported, it is public, and there is no point in making it public-public.
Exporting private APIs seems like an anti-pattern.

Reply:
Would prefer to follow Go conventions more than Geth conventions. Found several places showing how Geth developers haven't been warm with Go conventions, e.g. when naming getters `Get...`.

Reply:
@mandrigin so it's actually not entirely "private". It used by `github.com/ethereum/go-ethereum/rpc` package. Here `rpc` is a consumer of this exported PublicAPI type:
```go
func (s *MailService) APIs() []rpc.API {
    return []rpc.API{
        {
            Namespace: "shh",
            Version:   "1.0",
            Service:   NewPublicAPI(s.provider),
            Public:    true,
        },
    }
}
```

Reply:
@divan yeah, I just was talking about redundancy of the word `Public` in a publically exported type 😆 
It only makes sense if you have publically exported type called `PrivateAPI` somewhere, but that is quite a red flag for me.

Reply:
Anyway, that was the most minor of my comments. I'm not strictly opposed to that, it just looks weird to me.

Reply:
So that's what I'm trying to explain, that `Public` vs `Private` are naming schemes from geth's RPC subsystem and have nothing to do with language level public/private aka exported/unexported types. It means that methods of these API will be available either for public or private geth RPC endpoint.
And, as it follows already familiar naming scheme from geth codebase, this name is actually helpful to indicate that this type is a "Public" API in geth RPC terms.

Reply:
@divan thanks for the explanation! TIL 👍

Reply:
Just my 2c, ipc and inproc APIs are called `Private` in geth codebase. `Public` is meant to be available via HTTP, WS etc.